src/geometry.rs

```diff
@@ -89,11 +89,12 @@ pub trait Geom<'a>:
     /// ```
     /// use geos::{Geom, Geometry};
     ///
-    /// let geom = Geometry::new_from_wkt("POLYGON((0 0, 1 1, 1 2, 1 1, 0 0))")
+    /// // Bowtie polygon with self-intersection
```

Comment:
The problem with the original is that it shared two edges rather than self-intersecting at a point, so it broke because different versions of GEOS reported different parts of those shared edges as the self-intersection.

This uses the classic "bowtie" self-intersection shape with a single point of self-intersection.